src/types.jl

```diff
@@ -213,23 +505,26 @@ mutable struct RObject{S<:Sxp}
         r
     end
     # SymSxps are not garbage collected, so preserve not necessary.
-    RObject{S}(p::Ptr{SymSxp}) where S = new{S}(p)
+    RObject{SymSxp}(p::Ptr{SymSxp}) = new{SymSxp}(p)
```

Comment:
This isn't functionally the same as what was written but I'm not sure whether what was written before was intentional. It was implicitly relying on `convert(Ptr{S}, p::Ptr{SymSxp})` but I don't know enough about how this package works to know whether that's a sensible conversion to do. With this change, `RObject{S}(::Ptr{SymSxp})` for `S != SymSxp` will be a `MethodError`, won't it?

Reply:
Yeah, I think it will be a method error, but I think that having a method error there makes sense. The usual call would be `RObject(p)`, which invokes this definition:

```julia
RObject(p::Ptr{S}) where S<:Sxp = RObject{S}(p)
```

and thus would always use the narrowest possible construction.